File: presentation/src/main/java/com/everis/authspike/navigator/Navigator.java
package com.everis.authspike.navigator;

import android.app.Activity;
import android.content.Intent;

import com.everis.authspike.view.activity.ContactsActivity;
import com.everis.authspike.view.activity.HomeActivity;
import com.everis.authspike.view.activity.RegisterActivity;

/**
 * Created by everis on 25/04/18.
 */

public class Navigator {

    public Navigator() {
    }

    public void navigateToHomeActivity(Activity activity){
        Intent homeIntent = new Intent(activity, HomeActivity.class);
        activity.startActivity(homeIntent);
        activity.finish();
    }

    public void navigateToRegisterActivity(Activity activity){
        Intent registerIntent = new Intent(activity, RegisterActivity.class);
        activity.startActivity(registerIntent);
        activity.finish();
    }

    public void navigateToContactsActivity(Activity activity) {
        Intent contactsIntent = new Intent(activity, ContactsActivity.class);
        activity.startActivity(contactsIntent);
    }
}

File: presentation/src/main/java/com/everis/authspike/navigator/Navigator.kt
package com.everis.authspike.navigator

import android.content.Intent
import com.everis.authspike.R
import com.everis.authspike.view.activity.BaseActivity
import com.everis.authspike.view.activity.HomeActivity
import com.everis.authspike.view.activity.WelcomeActivity
import com.everis.authspike.view.fragment.ContactFragment
import com.everis.authspike.view.fragment.LoginFragment
import com.everis.authspike.view.fragment.RegisterFragment

class Navigator (private val activity: BaseActivity) {

    fun navigateToHomeActivity() {
        val homeIntent = Intent(activity, HomeActivity::class.java)
        activity.startActivity(homeIntent)
        activity.finish()
    }


    fun navigateToRegisterActivity() {
        val registerIntent = Intent(activity, WelcomeActivity::class.java)
        activity.startActivity(registerIntent)
        activity.finish()
    }

    fun navigateToContactsFragment() {
        val transaction = activity.supportFragmentManager.beginTransaction()
        transaction.replace(R.id.fragmentContainer, ContactFragment.newInstance()).commit()
    }

    fun navigateToRegisterFragment() {
        activity.supportFragmentManager.beginTransaction().replace(R.id.welcome_container, RegisterFragment.newInstance()).addToBackStack(RegisterFragment::class.java.simpleName).commit()
    }

    fun navigateToLoginFragment(activity: WelcomeActivity) {
        activity.supportFragmentManager.beginTransaction().replace(R.id.welcome_container, LoginFragment.newInstance(activity)).commit()
    }
}